### PR TITLE
Get aggregate state

### PR DESCRIPTION
**Pull request by fahchen**
This PR addresses https://github.com/commanded/commanded/issues/336

### CODE REVIEW

File: test/aggregates/aggregate_state_test.exs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+             }
+  end
+
+  # Assert aggregate's version equals the given expected version.
+  defp assert_aggregate_version(aggregate_uuid, expected_version) do
+    assert Aggregate.aggregate_version(DefaultApp, @aggregate_module, aggregate_uuid) ==
+             expected_version
+  end
+
+  # Restart the aggregate process
```
Comment (vasspilka) [inline test/aggregates/aggregate_state_test.exs:76]:
```suggestion
  # Shutdowns the aggregate process
```
The process does not automatically restart, right?

**Review by vasspilka — APPROVED** [review]
Nice! I actually was looking forward to this, there are cases I want to get to aggregate state directly, in my company we needed to make sure projections and aggregate state match when doing a big refactoring of the MVP which was based on projections being the "source of truth". 

I tested this on my pet project and seems to be working nicely :) . Awesome stuff!

**Comment by fahchen** [comment]
@vasspilka Thank you! I pushed up your requested changes.

**Comment by fahchen** [comment]
@slashdotdash could you have a look at this PR ?

**Comment by slashdotdash** [comment]
Thanks for the pull request @fahchen.

**Comment by stefanluptak** [comment]
I know I am bit late to this, but does this new functionality differ somehow from doing this?

```elixir
{:ok, ^aggregate_id} =
      Commanded.Aggregates.Supervisor.open_aggregate(
        My.Commanded.Application
        MyAggregate,
        aggregate_id
      )

    %__MODULE__{} =
      Commanded.Aggregates.Aggregate.aggregate_state(
        My.Commanded.Application,
        MyAggregate,
        aggregate_id
      )
```

It looks the same to me.

**Comment by fahchen** [comment]
> I know I am bit late to this, but does this new functionality differ somehow from doing this?
> 
> 
> 
> ```elixir
> 
> {:ok, ^aggregate_id} =
> 
>   Commanded.Aggregates.Supervisor.open_aggregate(
> 
>     My.Commanded.Application
> 
>     MyAggregate,
> 
>     aggregate_id
> 
>   )
> 
> 
> 
> %__MODULE__{} =
> 
>   Commanded.Aggregates.Aggregate.aggregate_state(
> 
>     My.Commanded.Application,
> 
>     MyAggregate,
> 
>     aggregate_id
> 
>   )
> 
> ```
> 
> 
> 
> It looks the same to me.

aggregate_state never fails, when the aggregate doesn't exist, the initial state is returned (For keeping it same with the original implementation at aggregate module).

But recently, I found this returning isn't so convenient, maybe we should return error tuple when the aggregate doesn't exist?

**Comment by stefanluptak** [comment]
> But recently, I found this returning isn't so convenient, maybe we should return error tuple when the aggregate doesn't exist?

Thanks a lot for your answer.
Regarding your question, I am not sure. Maybe I'll have an opinion after using your functionality more.

**Pull request merged**